src/hooks/usePhaseData.ts

```diff
@@ -0,0 +1,104 @@
+import { useCallback, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { LoadConstantsQuery } from 'src/components/Constants/LoadConstants.generated';
+import { useApiConstants } from 'src/components/Constants/UseApiConstants';
+import {
+  ActivityTypeEnum,
+  Phase,
+  PhaseEnum,
+} from 'src/graphql/types.generated';
+import { getLocalizedPhase } from 'src/utils/functions/getLocalizedPhase';
+import { getLocalizedTaskType } from 'src/utils/functions/getLocalizedTaskType';
+
+export type SetPhaseId = (activity: PhaseEnum | null) => void;
+export type Constants = LoadConstantsQuery['constant'] | undefined;
+
+type GetPhaseData = {
+  phaseData: Phase | null;
+  setPhaseId: SetPhaseId;
+  constants: LoadConstantsQuery['constant'] | undefined;
+  taskPhases: PhaseEnum[];
+  activityTypes: Map<ActivityTypeEnum, ActivityData>;
+  activitiesByPhase: Map<PhaseEnum, ActivityTypeEnum[]>;
+};
+
+export type ActivityData = {
+  name: string;
+  phaseId: PhaseEnum;
+  phase: string;
+  title?: string;
+};
+
+const phaseFromActivity = (
+  activity: PhaseEnum | null,
+  constants: LoadConstantsQuery['constant'] | undefined,
+): Phase | null => {
+  const phases = constants?.phases;
+  if (!activity || !phases) {
+    return null;
+  }
+  return (
+    phases.find((phase) => phase.id.toLowerCase() === activity.toLowerCase()) ??
+    null
+  );
+};
+
+export const usePhaseData = (phaseEnum?: PhaseEnum | null): GetPhaseData => {
```

Comment:
Again, at least tests and possibly comments showing the structure of the maps returned would be helpful here.

Reply:
I'm working on this one.